lib/src/rules/prefer_const_literals_to_create_immutables.dart

```diff
@@ -135,4 +111,25 @@ class _Visitor extends SimpleAstVisitor<void> {
       rule.reportLint(literal);
     }
   }
+
+  // todo(pq): consider making this a utility and sharing w/ `avoid_equals_and_hash_code_on_mutable_classes`
+  static bool _hasImmutableAnnotation(DartType? type) {
+    if (type is! InterfaceType) {
+      // This happens when we find an instance creation expression for a class
+      // that cannot be resolved.
+      return false;
+    }
+
+    InterfaceType? current = type;
+    // TODO(a14n) the is check looks unnecessary but prevents https://github.com/dart-lang/sdk/issues/33210
```

Comment:
I don't understand what `is` check this is referring to. If the comment is stale, consider updating or removing it.

Reply:
Good catch!  Removed.